plugins/prometheus/pkg/data/data_test.go

```diff
@@ -328,3 +329,151 @@ func TestAdjustMetricsForMultiContainerPods(t *testing.T) {
+	adjustedMetrics := adjustMetricsForMultiContainerPods(metrics, workloads)
+	assert.Equal(t, 9, len(adjustedMetrics), "number of metrics after adjusting per-pod metrics to be per-container")
+
+	// The first 3 metrics should be for the containers of pod1.
+	// Dynamically construct the expected container name (container1...,
+	// container2...) using the index `i`.
+	for i := 0; i <= 2; i++ {
+		assert.Equal(t,
+			model.Metric{
+				"namespace": "default",
+				"pod":       "pod1",
+				"container": model.LabelValue(fmt.Sprintf("container%dforpod1", i+1)), // container name "containerXforpod1"
+			},
+			adjustedMetrics[i].Metric,
+			fmt.Sprintf("labels for adjusted metric at index %d include the correct container name", i))
+	}
+
+	// Metrics at index 3-5 should be for the containers of pod2.
+	// Dynamically construct the expected container name (container1...,
+	// container2...) using the index `i`.
+	for i := 3; i <= 5; i++ {
+		assert.Equal(t,
+			model.Metric{
+				"namespace": "default",
+				"pod":       "pod2",
+				"container": model.LabelValue(fmt.Sprintf("container%dforpod2", i-2)), // container name "containerXforpod2"
+			},
+			adjustedMetrics[i].Metric,
+			fmt.Sprintf("labels for adjusted metric at index %d include the correct container name", i))
+	}
+
+	// Metrics at index 6-8 should be for the containers of pod3.
+	// Dynamically construct the expected container name (container1...,
+	// container2...) using the index `i`.
+	for i := 6; i <= 8; i++ {
+		assert.Equal(t,
+			model.Metric{
+				"namespace": "default",
+				"pod":       "pod3",
+				"container": model.LabelValue(fmt.Sprintf("container%dforpod3", i-5)), // container name "containerXforpod2"
+			},
+			adjustedMetrics[i].Metric,
+			fmt.Sprintf("labels for adjusted metric at index %d include the correct container name", i))
+	}
+
+	// Verify the capacity of the shared PVC was split up correctly
+	// Verify the capacity of the shared PVC was split up correctly per-pod and
+	// per-metric.
+	// Although the below only asserts values for a single pod (pod1), the
+	// correctness of the metric values does implicitly validate that the PVC has
+	// been split correctly across the pods that share the PVC.
+	// Additionally, all values of a single metric do not need to be checked, because PVC
+	// metrics use the same value for all time-stamps.
+	// Note that adjustedMetrics[0] has a slightly larger value than
+	// adjustedMetrics[1] or ...[2], because the first metric value holds the
+	// remainder of the divided value.
+	assert.Equal(t, model.SampleValue(954437178.0), adjustedMetrics[0].Values[0].Value, "the metric value for pod1 and container name container1forpod1")
+	assert.Equal(t, model.SampleValue(954437177.0), adjustedMetrics[1].Values[0].Value, "the metric value for pod1 and container name container2forpod1")
+	assert.Equal(t, model.SampleValue(954437177.0), adjustedMetrics[1].Values[0].Value, "the metric value for pod1 and container name container2forpod1")
```

Comment:
Assertions on lines `477` and `478` are duplicated

Reply:
Thank you - I've fixed this to now test the 3rd container.